include/itkCommandExhaustiveLog.h

```diff
@@ -0,0 +1,177 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkCommandExhaustiveLog_h
+#define itkCommandExhaustiveLog_h
+
+#include "itkMacro.h"
+#include "itkCommand.h"
+#include "itkImage.h"
+#include "itkExhaustiveOptimizerv4.h"
+
+namespace itk
+{
+/**
+ *\class CommandExhaustiveLog
+ *  \brief Monitors ExhaustiveOptimizerv4 and provides the search space as an image.
+ *
+ * This class is intended to assist a user in evaluating optimization performance
+ * over some transform search space. An itk::CommandExhaustiveLog object may be added
+ * to consume event data from an itk::ExhaustiveOptimizerv4 object as the optimizer 
+ * iterates over a pre-set domain of transform parameters and returns a metric 
+ * value at each point. The CommandExhaustiveLog observer object logs each 
+ * data point internally in an itk::Image object so that the parametric surface 
+ * is available after optimization concludes.
+ *
+ * A CommandExhaustiveLog object paired with an ExhaustiveOptimizerv4 may be
+ * useful for obtaining data to plan future optimization attempts, such as whether
+ * a region appears rough or curved in a certain way that may render gradient descent
+ * less effective. Exhaustive data may also help inform the user in regards to
+ * the performance of ongoing optimization, such as overlaying optimizer steps
+ * onto an image of the exhaustive parametric region in order to determine whether
+ * the optimizer learning rate is appropriate for the region.
+ * 
+ * Wrapping for this class is limited to available wrappings for the internal
+ * itk::Image data array. Users wishing to make use of transforms with large
+ * parameter lists may find it necessary to build ITK with custom wrappings to
+ * meet their needs.
+ *
+ * Template parameters for class CommandExhaustiveLog:
+ *
+ * - TInternalData = Element type stored at each location in the data image.
+ * - TImageDimension = Dimension of image equal to number of transform parameters
+ *
+ * \ingroup ITKOptimizationMonitor
+ */
+template <typename TInternalData, unsigned int TImageDimension>
```

Comment:
To be closer to ITK naming conventions and to be more specific, perhaps `TInternalData` -> `TValue`.